### ISSUE TITLE
Run-ALValidation: error AL1033: An error occurred while loading the included rule set file 

### ISSUE THREAD

**Issue by ti-jalopez** [microsoft/navcontainerhelper#1689]
**Describe the issue**
Run-ALValidation trhows error AL1033: An error occurred while loading the included rule set file.
Some moths ago, I've run Run-ALValidation with same parameters and app and it ran fine.

**Scripts used to create container and cause the issue**
```
$myLicense = 'C:\myLicense.flf'
$previousApps = @()
$apps = @('c:\myApp_1.0.0.30.app');
$affixes = 'MYPREFIX'

#Current Version
$validationResults = Run-AlValidation `
    -licenseFile $myLicense `
    -previousApps $previousApps `
    -apps $apps `
    -validateCurrent `
    -countries 'ES' `
    -affixes $affixes `
    -supportedCountries 'ES' `
    -verbose

```

**Full output of scripts**
```
  _____                               _                
 |  __ \                             | |               
 | |__) |_ _ _ __ __ _ _ __ ___   ___| |_ ___ _ __ ___ 
 |  ___/ _` | '__/ _` | '_ ` _ \ / _ \ __/ _ \ '__/ __|
 | |  | (_| | | | (_| | | | | | |  __/ |_  __/ |  \__ \
 |_|   \__,_|_|  \__,_|_| |_| |_|\___|\__\___|_|  |___/

Container name               bcserver
Image name                   
Credential                   admin/Bido8184
License file                 Specified
MemoryLimit                  8G
validateVersion              
validateCurrent              True
validateNextMinor            False
validateNextMajor            False
SasToken                     Not Specified
countries                    es
validateCountries            es
affixes                      MYPREFIX
supportedCountries           es
vsixFile                     
Install Apps
- None
Previous Apps
- None
Apps
- c:\myApp_1.0.0.30.app

  _____       _ _ _                                          _        _                            
 |  __ \     | | (_)                                        (_)      (_)                           
 | |__) |   _| | |_ _ __   __ _    __ _  ___ _ __   ___ _ __ _  ___   _ _ __ ___   __ _  __ _  ___ 
 |  ___/ | | | | | | '_ \ / _` |  / _` |/ _ \ '_ \ / _ \ '__| |/ __| | | '_ ` _ \ / _` |/ _` |/ _ \
 | |   | |_| | | | | | | | (_| | | (_| |  __/ | | |  __/ |  | | (__  | | | | | | | (_| | (_| |  __/
 |_|    \__,_|_|_|_|_| |_|\__, |  \__, |\___|_| |_|\___|_|  |_|\___| |_|_| |_| |_|\__,_|\__, |\___|
                           __/ |   __/ |                                                 __/ |     
                          |___/   |___/                                                 |___/      

Pulling mcr.microsoft.com/businesscentral:10.0.19042.746

Pulling generic image took 1 seconds
  _____       _                      _                         _   _  __           _       
 |  __ \     | |                    (_)                       | | (_)/ _|         | |      
 | |  | | ___| |_ ___ _ __ _ __ ___  _ _ __   ___    __ _ _ __| |_ _| |_ __ _  ___| |_ ___ 
 | |  | |/ _ \ __/ _ \ '__| '_ ` _ \| | '_ \ / _ \  / _` | '__| __| |  _/ _` |/ __| __/ __|
 | |__| |  __/ |_  __/ |  | | | | | | | | | |  __/ | (_| | |  | |_| | || (_| | (__| |_\__ \
 |_____/ \___|\__\___|_|  |_| |_| |_|_|_| |_|\___|  \__,_|_|   \__|_|_| \__,_|\___|\__|___/
                                                                                           
DETALLADO: Invoke-RestMethod -Method Get -Uri https://bcartifacts.azureedge.net/sandbox/?comp=list&restype=container
DETALLADO: GET https://bcartifacts.azureedge.net/sandbox/?comp=list&restype=container with 0-byte payload
...
DETALLADO: received -1-byte response of content type application/xml
Using https://bcartifacts.azureedge.net/sandbox/17.4.21491.21867/es

   _____                _   _                               _        _                 
  / ____|              | | (_)                             | |      (_)                
 | |     _ __ ___  __ _| |_ _ _ __   __ _    ___ ___  _ __ | |_ __ _ _ _ __   ___ _ __ 
 | |    | '__/ _ \/ _` | __| | '_ \ / _` |  / __/ _ \| '_ \| __/ _` | | '_ \ / _ \ '__|
 | |____| | |  __/ (_| | |_| | | | | (_| | | (__ (_) | | | | |_ (_| | | | | |  __/ |   
  \_____|_|  \___|\__,_|\__|_|_| |_|\__, |  \___\___/|_| |_|\__\__,_|_|_| |_|\___|_|   
                                     __/ |                                             
                                    |___/                                              

Creating container for country es
DETALLADO: Realice la operación 'Enumerar instancias de CIM' con los parámetros siguientes, ''namespaceName' = root\cimv2,'className' = Win32_OperatingSystem'.
DETALLADO: La operación 'Enumerar instancias de CIM' se completó.
DETALLADO: Realice la operación 'Enumerar instancias de CIM' con los parámetros siguientes, ''namespaceName' = root\cimv2,'className' = Win32_OperatingSystem'.
DETALLADO: La operación 'Enumerar instancias de CIM' se completó.
BcContainerHelper is version 2.0.4
BcContainerHelper is running as administrator
Host is Microsoft Windows 10 Pro - 20H2
Docker Client Version is 20.10.2
Docker Server Version is 20.10.2
Fetching all docker images
Using image mcr.microsoft.com/businesscentral:10.0.19042.746
Creating Container bcserver
Version: 17.4.21491.21867-ES
Style: sandbox
Multitenant: No
Platform: 17.0.21485.21847
Generic Tag: 1.0.1.3
Container OS Version: 10.0.19042.746 (20H2)
Host OS Version: 10.0.19042.789 (20H2)
Using hyperv isolation
Using locale es-ES
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Using license file C:\myLicense.flf
Additional Parameters:
--env customNavSettings=EnableTaskScheduler=True
Files in C:\ContainerHelper\Extensions\bcserver\my:
- AdditionalOutput.ps1
- license.flf
- MainLoop.ps1
- SetupVariables.ps1
- updatehosts.ps1
Creating container bcserver from image mcr.microsoft.com/businesscentral:10.0.19042.746
8fa87033c98128cb3a8038ce5ccdd44a3588d907caec23cd3c427150e6458847
Waiting for container bcserver to be ready
Using artifactUrl https://bcartifacts.azureedge.net/sandbox/17.4.21491.21867/es
Using installer from C:\Run\150-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
Copying PowerShell Scripts
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Determining Database Collation from c:\dl\sandbox\17.4.21491.21867\es\BusinessCentral-ES.bak
Changing Database Server Collation to Modern_Spanish_100_CS_AS
Restoring CRONUS Demo Database
Setting CompatibilityLevel for CRONUS on localhost\SQLEXPRESS
Modifying Business Central Service Tier Config File for Docker
Creating Business Central Service Tier
Installing SIP crypto provider: 'C:\Windows\System32\NavSip.dll'
Copying Web Client Files
Copying Client Files
Copying ModernDev Files
Copying additional files
Copying ConfigurationPackages
Copying Test Assemblies
Copying Extensions
Copying Applications
Copying Applications.ES
Starting Business Central Service Tier
Importing license file
Stopping Business Central Service Tier
Installation took 253 seconds
Installation complete
Initializing...
Setting host.docker.internal to 192.168.0.24 in container hosts file (copy from host hosts file)
Setting gateway.docker.internal to 192.168.0.24 in container hosts file (copy from host hosts file)
Setting kubernetes.docker.internal to 127.0.0.1 in container hosts file (copy from host hosts file)
Setting host.containerhelper.internal to 172.23.16.1 in container hosts file
Starting Container
Hostname is bcserver
PublicDnsName is bcserver
WARNING: Container starts with TimeZone = Hora estÃ¡ndar Europa Occidental, which is not recognized in the list of TimeZones.
Using NavUserPassword Authentication
Creating Self Signed Certificate
Self Signed Certificate Thumbprint 604BA01052D4E8893D7666D3DF5A75CD1E1FEE60
Modifying Service Tier Config File with Instance Specific Settings
Modifying Service Tier Config File with settings from environment variable
Setting EnableTaskScheduler to True
Starting Service Tier
Registering event sources
Creating DotNetCore Web Server Instance
Enabling Financials User Experience
Using license file 'c:\run\my\license.flf'
Import License
Creating http download site
Setting SA Password and enabling SA
Creating admin as SQL User and add to sysadmin
Creating SUPER user
Container IP Address: 172.23.22.156
Container Hostname  : bcserver
Container Dns Name  : bcserver
Web Client          : http://bcserver/BC/
Dev. Server         : http://bcserver
Dev. ServerInstance : BC
Setting bcserver to 172.23.22.156 in host hosts file

Files:
http://bcserver:8080/ALLanguage.vsix

Container Total Physical Memory is 8.5Gb
Container Free Physical Memory is 6.0Gb

Initialization took 50 seconds
Ready for connections!
Reading CustomSettings.config from bcserver
Creating Desktop Shortcuts for bcserver
Container bcserver successfully created

Use:
Get-BcContainerEventLog -containerName bcserver to retrieve a snapshot of the event log from the container
Get-BcContainerDebugInfo -containerName bcserver to get debug information about the container
Enter-BcContainer -containerName bcserver to open a PowerShell prompt inside the container
Remove-BcContainer -containerName bcserver to remove the container again
docker logs bcserver to retrieve information about URL's again

Creating container took 316 seconds
  _____                   _                                    _____                           _____            
 |  __ \                 (_)                 /\               / ____|                         / ____|           
 | |__) |   _ _ __  _ __  _ _ __   __ _     /  \   _ __  _ __| (___   ___  _   _ _ __ ___ ___| |     ___  _ __  
 |  _  / | | | '_ \| '_ \| | '_ \ / _` |   / /\ \ | '_ \| '_ \\___ \ / _ \| | | | '__/ __/ _ \ |    / _ \| '_ \ 
 | | \ \ |_| | | | | | | | | | | | (_| |  / ____ \| |_) | |_) |___) | (_) | |_| | | | (__  __/ |____ (_) | |_) |
 |_|  \_\__,_|_| |_|_| |_|_|_| |_|\__, | /_/    \_\ .__/| .__/_____/ \___/ \__,_|_|  \___\___|\_____\___/| .__/ 
                                   __/ |          | |   | |                                              | |    
                                  |___/           |_|   |_|                                              |_|    
Extracting C:\ContainerHelper\4263d630-ca5d-4e5a-bed7-2836537f358b\myApp_1.0.0.30.app
Analyzing: myApp_1.0.0.30.app
Using affixes: MYPREFIX
Using supportedCountries: es
Downloading C:\ContainerHelper\3cde66e8-0ae5-46c1-82d3-1609f1496a5b\appsource.default.ruleset.json
AppSourceCop.json content:
{
    "mandatoryAffixes":  [
                             "MYPREFIX"
                         ],
    "supportedCountries":  [
                               "es"
                           ]
}
Ruleset.json content:
{
    "name":  "Run-AlCops RuleSet",
    "description":  "Generated by Run-AlCops",
    "includedRuleSets":  [
                             {
                                 "path":  "C:\\ContainerHelper\\3cde66e8-0ae5-46c1-82d3-1609f1496a5b\\appsource.default.ruleset.json",
                                 "action":  "Default"
                             }
                         ]
}
Using Symbols Folder: C:\ContainerHelper\401db574-9b4f-42b3-9cbe-5e54d10a924a
Copying System.app from Container
Copying Microsoft_Application_17.4.21491.21867.app from Container
Copying Microsoft_Base Application_17.4.21491.21867.app from Container
Copying Microsoft_System Application_17.4.21491.21867.app from Container
Compiling...
.\alc.exe /project:"C:\ProgramData\BcContainerHelper\3cde66e8-0ae5-46c1-82d3-1609f1496a5b" /packagecachepath:"C:\ProgramData\BcContainerHelper\401db574-9b4f-42b3-9cbe-5e54d10a924a" /out:"C:\P
rogramData\BcContainerHelper\3cde66e8-0ae5-46c1-82d3-1609f1496a5b\myApp_1.0.0.30.app" /analyzer:C:\build\vsix\extension\bin\Analyzers\Microsoft.Dynamics.Nav.AppS
ourceCop.dll /ruleset:C:\ProgramData\BcContainerHelper\3cde66e8-0ae5-46c1-82d3-1609f1496a5b\ruleset.json /assemblyprobingpaths:"C:\Program Files (x86)\Microsoft Dynamics NAV\170\RoleTailored 
Client","C:\Program Files\Microsoft Dynamics NAV\170\Service","C:\Program Files (x86)\Open XML SDK\V2.5\lib","c:\Windows\Microsoft.NET\Assembly","C:\Test Assemblies\Mock Assemblies"
Microsoft (R) AL Compiler version 6.4.6.9919
Copyright (C) Microsoft Corporation. All rights reserved

error AL1033: An error occurred while loading the included rule set file C:\ProgramData\BcContainerHelper\3cde66e8-0ae5-46c1-82d3-1609f1496a5b\ruleset.json - Could not find a part of the path
 'C:\ContainerHelper\3cde66e8-0ae5-46c1-82d3-1609f1496a5b\appsource.default.ruleset.json'.
Unexpected error while validating app. Error is: App generation failed
en Compile-AppInBcContainer, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.4\AppHandling\Compile-AppInNavContainer.ps1: línea 515
en <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.4\AppHandling\Run-AlCops.ps1: línea 68
en <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.4\AppHandling\Run-AlCops.ps1: línea 210
en Run-AlCops, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.4\AppHandling\Run-AlCops.ps1: línea 112
en <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.4\AppHandling\Run-AlValidation.ps1: línea 495
en <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.4\AppHandling\Run-AlValidation.ps1: línea 478
en <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.4\AppHandling\Run-AlValidation.ps1: línea 393
en <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.4\AppHandling\Run-AlValidation.ps1: línea 367
en Run-AlValidation, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.4\AppHandling\Run-AlValidation.ps1: línea 365
en <ScriptBlock>, <sin archivo>: línea 11

  _____                           _                _____            _        _                 
 |  __ \                         (_)              / ____|          | |      (_)                
 | |__) |___ _ __ ___   _____   ___ _ __   __ _  | |     ___  _ __ | |_ __ _ _ _ __   ___ _ __ 
 |  _  // _ \ '_ ` _ \ / _ \ \ / / | '_ \ / _` | | |    / _ \| '_ \| __/ _` | | '_ \ / _ \ '__|
 | | \ \  __/ | | | | | (_) \ V /| | | | | (_| | | |____ (_) | | | | |_ (_| | | | | |  __/ |   
 |_|  \_\___|_| |_| |_|\___/ \_/ |_|_| |_|\__, |  \_____\___/|_| |_|\__\__,_|_|_| |_|\___|_|   
                                           __/ |                                               
                                          |___/                                                

Removing container bcserver
Removing bcserver from host hosts file
Removing bcserver-* from host hosts file
Removing C:\ContainerHelper\Extensions\bcserver

Removing container took 33 seconds

AL Validation finished in 396 seconds
 __      __   _ _     _       _   _               _____                _ _       
 \ \    / /  | (_)   | |     | | (_)             |  __ \              | | |      
  \ \  / /_ _| |_  __| | __ _| |_ _  ___  _ __   | |__) |___ ___ _   _| | |_ ___ 
   \ \/ / _` | | |/ _` |/ _` | __| |/ _ \| '_ \  |  _  // _ \ __| | | | | __/ __|
    \  / (_| | | | (_| | (_| | |_| | (_) | | | | | | \ \  __\__ \ |_| | | |_\__ \
     \/ \__,_|_|_|\__,_|\__,_|\__|_|\___/|_| |_| |_|  \_\___|___/\__,_|_|\__|___/

Unexpected error while validating app. Error is: App generation failed
  _____                          ___      __   _ _     _       _   _               ______    _ _                
 |  __ \                   /\   | \ \    / /  | (_)   | |     | | (_)             |  ____|  (_) |               
 | |__) |   _ _ __ ______ /  \  | |\ \  / /_ _| |_  __| | __ _| |_ _  ___  _ __   | |__ __ _ _| |_   _ _ __ ___ 
 |  _  / | | | '_ \______/ /\ \ | | \ \/ / _` | | |/ _` |/ _` | __| |/ _ \| '_ \  |  __/ _` | | | | | | '__/ _ \
 | | \ \ |_| | | | |    / ____ \| |  \  / (_| | | | (_| | (_| | |_| | (_) | | | | | | | (_| | | | |_| | | |  __/
 |_|  \_\__,_|_| |_|   /_/    \_\_|   \/ \__,_|_|_|\__,_|\__,_|\__|_|\___/|_| |_| |_|  \__,_|_|_|\__,_|_|  \___|
...
```

**Screenshots**

**Additional context**
- does it happen all the time?
- did it use to work?

**Comment by freddydk** [comment]
I think this is a bug occuring if you use a custom hosthelperfolder.
I will submit a fix for this.
If you need it before, you will need to use standard hosthelperfolder setting.

**Comment by ti-jalopez** [comment]
Ok, I can wait, so I'm using a custom hosthelperfolder.

**Comment by freddydk** [comment]
Fix shipped in 2.0.5

**Comment by ti-jalopez** [comment]
I've tested the new version and it runs fine
